### PR TITLE
For #7295 - Adds new custom ping: 'installation'

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
### Pull Request checklist

- [x] **Tests**: This PR includes unit tests.
- [x] **Screenshots**: This PR doesn't include screenshots because no UI changes have been made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
This will also need a data-review. You can copy the data review request from https://github.com/mozilla-mobile/fenix/pull/8064#issuecomment-581668781 as an example. Make sure to make the appropriate changes in the request form for what you are testing. Afterwards, add a needs:data-review label and probably ping @boek for the review.

**Comment by codrut-topliceanu** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) **What questions will you answer with this data?**
- Details about Adjust campaign

2) **Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?**
- This will help us prioritize bugs and features related to add-ons based on user need.

3) **What alternative methods did you consider to answer these questions? Why were they not sufficient?**
- No current alternative methods.

4) **Can current instrumentation answer these questions?**
- Currently there are no events to track this.

5) **List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data c](https://wiki.mozilla.org/Firefox/Data_Collection)[ollection ](https://wiki.mozilla.org/Firefox/Data_Collection)[categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki.**
- All data is Category 1.

6) **How long will this data be collected?**
- Until 2020-09-01

7) **What populations will you measure?**
-  All release, beta, and nightly users with telemetry enabled.

8) **Please provide a general description of how you will analyze this data.**
- Glean / Amplitude

9) **Where do you intend to share the results of your analysis?**
-  Only on Glean, Amplitude and with mobile teams.



@gabrielluong , I'd appreciate if you give my request a look and see if it's in order. 
@boek , could you please help with the review?

File: app/metrics.yaml [hunk 1]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
       - fenix-core@mozilla.com
     expires: "2020-09-01"
+
+installation:
+  campaign:
+   type: string
+   send_in_pings:
+     - installation
+   description: >
+     Campaign
```
Comment (Dexterp37) [inline app/metrics.yaml]: Please be more explicit and descriptive about what this is, its format and its purpose.

File: app/metrics.yaml [hunk 2]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+     - https://github.com/mozilla-mobile/fenix/pull/8074
+   notification_emails:
+     - fenix-core@mozilla.com
+   expires: "2020-09-01"
+  network:
+    type: string
+    send_in_pings:
+      - installation
+    description: >
+      Network
```
Comment (Dexterp37) [inline app/metrics.yaml]: Please be more explicit and descriptive about what this is, its format and its purpose.

File: app/metrics.yaml [hunk 3]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+      - https://github.com/mozilla-mobile/fenix/pull/8074
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-09-01"
+  adgroup:
+    type: string
+    send_in_pings:
+      - installation
+    description: >
+      AdGroup
```
Comment (Dexterp37) [inline app/metrics.yaml]: Please be more explicit and descriptive about what this is, its format and its purpose.

File: app/metrics.yaml [hunk 4]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+      - https://github.com/mozilla-mobile/fenix/pull/8074
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-09-01"
+  creative:
+    send_in_pings:
+      - installation
+    type: string
+    description: >
+      Creative
```
Comment (Dexterp37) [inline app/metrics.yaml]: Please be more explicit and descriptive about what this is, its format and its purpose.

File: app/metrics.yaml [hunk 5]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+      - https://github.com/mozilla-mobile/fenix/pull/8074
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-09-01"
+  timestamp:
+    send_in_pings:
+      - installation
+    type: string
+    description: >
+      Timestamp
```
Comment (Dexterp37) [inline app/metrics.yaml]: Please be more explicit and descriptive about what this is, its format and its purpose.

File: app/metrics.yaml [hunk 6]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+      - https://github.com/mozilla-mobile/fenix/issues/7295
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/8074
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-09-01"
+  timestamp:
+    send_in_pings:
+      - installation
+    type: string
```
Comment (Dexterp37) [inline app/metrics.yaml]: This metric is called "timestamp", but it has a string type. What's the type of data going in there?

File: app/metrics.yaml [hunk 7]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
     notification_emails:
       - fenix-core@mozilla.com
     expires: "2020-09-01"
+
+installation:
+  campaign:
+   type: string
```
Comment (Dexterp37) [inline app/metrics.yaml]: In order for me to help checking if this is the right type, can you provide an example for all these metrics?

File: app/metrics.yaml [hunk 8]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+    type: string
+    description: >
+      Timestamp
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/7295
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/8074
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-09-01"
```
Comment (Dexterp37) [inline app/metrics.yaml]: nit: missing newline at the end of the file

File: app/metrics.yaml [hunk 9]
```diff
@@ -1776,3 +1776,70 @@ app_theme:
...
+  timestamp:
+    send_in_pings:
+      - installation
+    type: string
+    description: >
+      Timestamp
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/7295
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/8074
```
Comment (Dexterp37) [inline app/metrics.yaml]: This needs to be a link to the *answer* of the data review request (also for the other metrics as well)

File: app/pings.yaml [hunk 10]
```diff
@@ -18,3 +18,14 @@ activation:
     - https://github.com/mozilla-mobile/fenix/pull/1707#issuecomment-486972209
   notification_emails:
     - fenix-core@mozilla.com
+
+installation:
+  description: >
+    Intended for counting user installs.
```
Comment (Dexterp37) [inline app/pings.yaml]: Can you please be more explicit about what this is, its purpose, etc.?

File: app/pings.yaml [hunk 11]
```diff
@@ -18,3 +18,14 @@ activation:
...
+  include_client_id: false
+  bugs:
+    - 7295
```
Comment (Dexterp37) [inline app/pings.yaml]: Please put the full link to the issue here

File: app/pings.yaml [hunk 12]
```diff
@@ -18,3 +18,14 @@ activation:
...
+  data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/1707#issuecomment-486972209
```
Comment (Dexterp37) [inline app/pings.yaml]: This data review points to another ping type. Please do request a data review for this new collection. See [here](https://wiki.mozilla.org/Firefox/Data_Collection).

File: app/metrics.yaml (hunk 6 above)
Comment (codrut-topliceanu) [reply app/metrics.yaml]: The issue did not specify a date-time format so currently I'm just sending an unix timestamp obtained from the device as a string. Please let me know what format is required for the field.

File: app/metrics.yaml (hunk 1 above)
Comment (codrut-topliceanu) [reply app/metrics.yaml]: In regards to this and all the other improper descriptions in the metrics.yaml & pings.yaml: 
- I'm afraid [the issue](https://github.com/mozilla-mobile/fenix/issues/7295) doesn't specify what those adjust properties represent, so I did not know what to write in their descriptions. I was hoping a reviewer would help.

File: app/metrics.yaml (hunk 7 above)
Comment (codrut-topliceanu) [reply app/metrics.yaml]: I did not find a way to trigger Adjust to populate these fields, I just used mock data to test them. So I cannot provide an example.

File: app/metrics.yaml (hunk 8 above)
Comment (codrut-topliceanu) [reply app/metrics.yaml]: will fix

File: app/metrics.yaml (hunk 9 above)
Comment (codrut-topliceanu) [reply app/metrics.yaml]: I will add a link to the answer when it will be available.

File: app/pings.yaml (hunk 11 above)
Comment (codrut-topliceanu) [reply app/pings.yaml]: will fix

File: app/pings.yaml (hunk 12 above)
Comment (codrut-topliceanu) [reply app/pings.yaml]: Will add link to the proper data review answer when one will be available

File: app/pings.yaml (hunk 10 above)
Comment (codrut-topliceanu) [reply app/pings.yaml]: Maybe someone from Glean team can provide some input regarding this?

File: app/metrics.yaml (hunk 1 above)
Comment (Dexterp37) [reply app/metrics.yaml]: I think @boek or somebody else on the Fenix team could probably provide more insights with this :)

File: app/metrics.yaml (hunk 6 above)
Comment (Dexterp37) [reply app/metrics.yaml]: The Glean SDK has a series of [metric types](https://mozilla.github.io/glean/book/user/metrics/index.html) that can be used to represent data. Strings can be a last resort choice (given that they make no assumption on the underlying data) only if other types do not work :) So this really depends on the shape of this data coming in.

If all that's required is a date/time of when this ping was submitted, then you don't need to add anything, it's already being reported in the [ping_info section](https://mozilla.github.io/glean/book/user/pings/index.html) of glean pings.

File: app/metrics.yaml (hunk 7 above)
Comment (Dexterp37) [reply app/metrics.yaml]: Did someone provide you with mock data from Adjust?

Comment (codrut-topliceanu) [reply app/metrics.yaml]: I'm afraid not.

File: app/pings.yaml (hunk 10 above)
Comment (Dexterp37) [reply app/pings.yaml]: I think that the person/owner of this ping (who ? Data Science? The Fenix Team?) should provide a reasonable description of why we're asking our users to share this data with us: what is this for and so on :)

File: app/metrics.yaml (hunk 7 above)
Comment (Dexterp37) [reply app/metrics.yaml]: All right, @boek , do you know what is the shape of the data we're trying to collect here?

**Comment by codrut-topliceanu** [comment]
@sblatz Could you please help with more details regarding [7295](https://github.com/mozilla-mobile/fenix/issues/7295)? (details about adgroup, adnetwork, timestamp format, etc.)

**Comment by boek** [comment]
placeholder

**Comment by mmccorks** [comment]
### **Data Review Form (to be filled by Data Stewards)**

 1.   Is there or will there be documentation that describes the schema for the ultimate data set in a public, complete, and accurate way?
    Yes, metrics.yaml and metrics.md

2.    Is there a control mechanism that allows the user to turn the data collection on and off?
    Yes, by opting out of telemetry.

3.    If the request is for permanent data collection, is there someone who will monitor the data over time?
    Not permanent data collection.

4.   Using the category system of data types on the Mozilla wiki, what collection type of data do the requested measurements fall under?
    Category 1 and Category 2 data.

5.   Is the data collection request for default-on or default-off?
    Default on.

6.    Does the instrumentation include the addition of any new identifiers (whether anonymous or otherwise; e.g., username, random IDs, etc. See the appendix for more details)?
    No new identifiers.

7.    Is the data collection covered by the existing Firefox privacy notice?
    Yes.

8.   Does there need to be a check-in in the future to determine whether to renew the data?
    Requires a check in at expiry (2020-09-01).

9.    Does the data collection use a third-party collection tool?
    No. 

----
data-review +

**Pull request merged**